src/java.base/share/classes/java/lang/reflect/Modifier.java

```diff
@@ -221,6 +231,22 @@ public static boolean isStrict(int mod) {
      * {@code toString} with the appropriate mask from a method like
      * {@link #constructorModifiers} or {@link #methodModifiers}.
      *
+     * @apiNote
+     * To make a high-fidelity representation of the Java source
+     * modifiers of a class or member, source-level modifiers that do
+     * not <em>not</em> have a constant in the this class should be
+     * included and ordered consistent with the full recommended
+     * ordering for that kind of declaration as given in <cite>The
+     * Java Language Specification</cite>. For example, for a
+     * {@linkplain Method#toGenericString() method} the "{@link
+     * Method#isDefault() default}" modifier is ordered immediately
+     * before "{@code static}" (JLS {@jls 9.4}). For a {@linkplain
```

Comment:
Nothing wrong with that, but note that unlike other similar sections, for whatever reason, JLS 9.4 does **not** suggest this:
> If two or more (distinct) ... modifiers appear in a ... declaration, it is customary, though not required, that they appear in the order consistent with that shown above in the production for ...

I [mused] about the reasons behind it recently, but only JLS experts know them for sure. (CC'in @dansmithcode)

[mused]: https://mail.openjdk.org/pipermail/core-libs-dev/2024-January/117398.html